### ISSUE TITLE
Support for digital gauges

### ISSUE THREAD

**Issue by DeeEmm** [DeeEmm/DIY-Flow-Bench#60]
Add support for standard digital gauge protocol to be able to record valve lift as suggested by Edvinas Juodviršis.

- Support standard serial gauge protocol.
- Lift value recorded and stored with flow data 
- Lift value displayed on dashboard

**Comment by DeeEmm** [comment]
Some information on serial gauge protocol - https://sites.google.com/site/marthalprojects/home/arduino/arduino-reads-digital-caliper

**Comment by DeeEmm** [comment]
Additional Gauge protocol information

https://hackaday.io/project/511-digital-dial-indicator-cnc-surface-probe

https://forum.arduino.cc/t/garant-digital-dial-indicator-encoder-digimatic-rs232-without-data-cable/543582

**Comment by DeeEmm** [comment]
Received ebay el-cheapo digital dial indicator that has serial data capabilities.

Not planning to take a look at this just yet as I need to finalise the shield design and testing but it's here and ready to go when I do.

![IMG_20210922_122933](https://user-images.githubusercontent.com/3038710/134276871-fa92fc12-1a45-4f51-b831-6161995497d8.jpg)

![IMG_20210922_122954](https://user-images.githubusercontent.com/3038710/134276877-c6ed7473-2c16-44a4-89c5-9e93d77ac84e.jpg)

![IMG_20210922_123014](https://user-images.githubusercontent.com/3038710/134276890-92be0341-fdce-45bd-b691-e414b735a572.jpg)

![IMG_20210922_123018](https://user-images.githubusercontent.com/3038710/134276903-e456ac3b-6305-4d91-9e17-8b8d84227684.jpg)

**Comment by baker27** [comment]
Do you need code for reading the dial indicator?

**Comment by DeeEmm** [comment]
If you have some or want to write some, that'd be great.

I was going to adapt the examples above

**Comment by KirikauKiwi** [comment]
@DeeEmm do you have the [serial cable](https://vi.aliexpress.com/item/1005006288776709.html) to plug in to your digital dial indicator?

It also looks like there are [USB versions](https://vi.aliexpress.com/item/1005002989551967.html) of the DDI too, the port on the DDI is USB and not a proprietary style like on the serial version.

I wonder if the data output is the same for the DDI's measuring 0.01mm and 0.001mm as the USB version has a 0.01 or 0.001 selector switch on the box in the middle of the USB cable.

**Comment by DeeEmm** [comment]
I don't recall. I suspect not. I seem to recall that I found the pin outs and intended to solder direct. But that looks like a bargain and worth it for the cable alone.

The USB version looks like it's using a UART Serial converter.

IIRC the intent was to use Serial 2 to connect the gauge, so the first style is more appropriate. 

Gauge[Serial] -> ESP32[serial 2]

The USB version would need to be converted back to straight serial to connect to serial 2 so you would need to add a USB host controller into the pipe...

Gauge[Serial] -> UART Slave -> UART Master -> ESP32[serial 2]

**Comment by DeeEmm** [comment]
> I wonder if the data output is the same for the DDI's measuring 0.01mm and 0.001mm as the USB version has a 0.01 or 0.001 selector switch on the box in the middle of the USB cable.

Not sure. I seem to recall that they follow a common standard, so suspect that they output would be the same. This is definitely true if they are using a UART as they are simply converting the data stream

**Comment by DeeEmm** [comment]
Bought a couple for 'research' purposes :laughing:

**Comment by stefan63** [comment]
Me thinking loud again.........
If we are using 1.8 deg (200 steps/rev) stepper motors driving a lead screw with 2mm pitch will gwt us (1 mm) / 800 = 
0,00125 mm  / revolution. Knowing this will make the messurement with a digital gauge unnecessary!
https://www.circuitist.com/how-to-calculate-steps-per-mm-lead-screw-and-gt2-timing-belt/

**Comment by DeeEmm** [comment]
> Me thinking loud again......... If we are using 1.8 deg (200 steps/rev) stepper motors driving a lead screw with 2mm pitch will get us (1 mm) / 800 = 0,00125 mm movement/step. Knowing this will make the messurement with a digital gauge unnecessary! https://www.circuitist.com/how-to-calculate-steps-per-mm-lead-screw-and-gt2-timing-belt/

Indeed. This is the exact idea. Also with automated valve adjustment and automated flow control it will be possible to map the entire valve lift range automatically.

**Comment by stefan63** [comment]
Maybe one of those could work, almost plug&play!
https://www.aliexpress.com/item/1005007160376705.html?

**Comment by stefan63** [comment]
There is also .9 degree steppers available ;)

Den sön 17 nov. 2024 22:29Mick Percy ***@***.***> skrev:
